plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -327,4 +331,59 @@ protected Session joinPushdownEnabled(Session session)
                 .setCatalogSessionProperty(session.getCatalog().orElseThrow(), "join_pushdown_enabled", "true")
                 .build();
     }
+
+    @Test(dataProvider = "testCaseInsensitivePredicatePushdownDataProvider")
+    public void testCaseInsensitivePredicatePushdown(String type)
+    {
+        Session useCaseInsensitivePredicatePushdown = Session.builder(getSession())
+                .setCatalogSessionProperty(getSession().getCatalog().orElseThrow(), UNSAFE_VARCHAR_PREDICATE_PUSHDOWN, "true")
+                .build();
+
+        if (hasBehavior(SUPPORTS_PREDICATE_PUSHDOWN_WITH_VARCHAR_INEQUALITY)) {
+            assertQueryFails(useCaseInsensitivePredicatePushdown, "SHOW SESSION", "Unknown session property .*.unsafe-varchar-predicate-pushdown");
+            return;
+        }
+        String tableName = "test_case_insensitive_predicate_pushdown" + randomTableSuffix();
+        assertQuerySucceeds(format("CREATE TABLE %s (value %s)", tableName, type));
+        assertQuerySucceeds(format("INSERT INTO %s VALUES 'a', 'A', 'b', 'B'", tableName));
+
+        assertThat(query(useCaseInsensitivePredicatePushdown, format("SELECT value FROM %s WHERE value = 'a'", tableName)))
+                // possibly incorrect results with unexpected 'A'
+                .isFullyPushedDown();
+        assertThat(query(useCaseInsensitivePredicatePushdown, format("SELECT value FROM %s WHERE value != 'a'", tableName)))
+                // possibly incorrect results with missing 'A'
+                .isFullyPushedDown();
+        assertThat(query(useCaseInsensitivePredicatePushdown, format("SELECT value FROM %s WHERE value > 'A'", tableName)))
+                // possibly incorrect results with missing 'a'
+                .isFullyPushedDown();
+
+        // Verify that results are correct when session property is not used
+        assertThat(query(format("SELECT value FROM %s WHERE value != 'a'", tableName)))
+                .skippingTypesCheck()
+                .matches("VALUES 'A', 'b', 'B'")
+                .isNotFullyPushedDown(FilterNode.class);
+
+        assertThat(query(format("SELECT value FROM %s WHERE value = 'a'", tableName)))
+                .skippingTypesCheck()
+                .matches("VALUES 'a'");
+        assertThat(query(format("SELECT value FROM %s WHERE value != 'a'", tableName)))
+                .isNotFullyPushedDown(FilterNode.class)
+                .skippingTypesCheck()
+                .matches("VALUES 'A', 'b', 'B'");
+        assertThat(query(format("SELECT value FROM %s WHERE value > 'A'", tableName)))
```

Comment:
I would prefer if you tested that ordering is correct too.

Some databases/collations can sort `A, a, B, b` or `a, b, A, B` or `A, B, a, b`.
Your test would pass for collations that sort as `A, B, a, b` too when I think it should fail.

Reply:
> Or am I broadening the scope by introducing ordering with predicate pushdown?

You are. However it sounds like a good idea to have a single toggle to control both pushdown at once.

Reply:
BTW same goes to aggregation push down. Notice that pushing down DISTINCT or GROUP BY on varchar column may produce different results. Is it handled today anyhow? It sounds to me that it could be also controlled with the same switch.

@findepi ?

Reply:
It should, but we also need a reusable test for aggregations as well.

Reply:
Reusable tests for order by and aggregations are out of my scope. Will create tickets for that, possibly as bugs as I would expect now wrong results. Also I am going to rename toggle from `unsafe-varchar-predicate-pushdown.enabled` to `unsafe-varchar-pushdown.enabled`.

Reply:
https://github.com/trinodb/trino/issues/7320, order by is handled by io.trino.plugin.jdbc.BaseJdbcConnectorTest#testCaseSensitiveTopNPushdown